### PR TITLE
feat: Release notes for Codacy Cloud March 2024

### PR DESCRIPTION
**Pull request by nicklem**
Adds auto-generated release notes for Codacy Cloud March 2024

### :eyes: Live preview
https://feat-release-notes-cloud-2024-03--docs-codacy.netlify.app/release-notes/cloud/cloud-2024-03/

### :construction: To do
- [x] Add the new release notes page to `mkdocs.yml` and `docs/release-notes/index.md`
- [x] Review generated release notes
- [x] Review list of issues with missing release notes
  - Awaiting squad reply [here](https://codacy.slack.com/archives/C03HSK48GDA/p1712157554938339), [here](https://codacy.slack.com/archives/CQESXTUHW/p1712157043537679) 
- [x] Review links to changelogs of updated tools
- [x] Update release date and remove `TODO` comments

### CODE REVIEW

**Review by stefanvacareanu7 — APPROVED** [review]

**Review by machadoit — APPROVED** [review]

File: docs/release-notes/cloud/cloud-2024-03.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+-   Streamlined new organization setup with a guided path for adding new organizations and a setup checklist on the [Organization overview](../../organizations/organization-overview.md) page. (ALA-825)
+-   It is now possible to display penetration testing results in the Codacy app. See how to [start pen testing with Codacy](https://go.codacy.com/pen-testing-product). (TAROT-2553)
+-   It's now possible to [unfollow a repository](../../organizations/managing-repositories.md#follow-unfollow) through the Codacy app. (PLUTO-902)
+-   In Enterprise plans where Codacy automatically manages seat usage for your organization, the billing period is now based only on the code analysis activity. Codacy app users who don’t contribute to your private repositories no longer occupy a seat. (PLUTO-867)
+
+## Bug fixes
+
+-   Fixed a bug that prevented Security and risk management items from linking to their respective Jira issues. (TAROT-2648)
+-   Fixed an issue where the repository update system was being too eager to time out some git operations that are used to sync the current state of a repository. (IO-1019)
```
Comment (claudiacarpinteiro) [inline docs/release-notes/cloud/cloud-2024-03.md:30]: Tweak for clarity.

```suggestion
-   Fixed an issue preventing pull requests from showing on Codacy due to Git operations timeout when synchronizing the current state of a repository. (IO-1019)
```

File: docs/release-notes/cloud/cloud-2024-03.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+-   @coffeelint/cli 5.2.11
+-   Ameba 1.5.0
+-   Bandit 1.7.5
+-   Brakeman 4.3.1
+-   bundler-audit (deprecated) 0.9.1
+-   **[Checkov 3.2.37](https://github.com/bridgecrewio/checkov/releases/tag/3.2.37) (updated from 3.0.25)**
+-   Checkstyle 10.13.0
+-   Clang-Tidy 10.0.1
+-   CodeNarc 3.3.0
```
Comment (claudiacarpinteiro) [inline docs/release-notes/cloud/cloud-2024-03.md:50]: Adding **CoffeeLint** 

```suggestion
-   CodeNarc 3.3.0
-   CoffeeLint 5.2.11
```

File: docs/release-notes/cloud/cloud-2024-03.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+## Feature removal
+
+-   We removed the option to bulk copy code patterns between repositories and to import code pattern configurations from another repository. If you want to share tool and code pattern configurations between repositories, the recommended approach is to use a [coding standard](../../organizations/using-coding-standards.md). (ALA-826)
+
+## Tool versions
+
+Codacy Cloud now includes the tool versions below. The tools that were recently updated are highlighted in bold:
+
+-   @coffeelint/cli 5.2.11
```
Comment (claudiacarpinteiro) [inline docs/release-notes/cloud/cloud-2024-03.md:42]: Add **CoffeeLint** below, instead...

```suggestion
```

**Review by claudiacarpinteiro — COMMENTED** [review]
Looks good! 👍 
I left some tweaks.

**Review by claudiacodacy — APPROVED** [review]

**Review by claudiacarpinteiro — APPROVED** [review]

**Pull request merged**